libsplinter/src/runtime/service/timer/thread.rs

```diff
@@ -0,0 +1,261 @@
+                                service_id, err
+                            );
+                            return;
+                        }
+                    }
+                };
+
+                if let Err(err) = handler.handle_timer(&*timer_sender, service_id) {
+                    error!("{}", err);
+                }
+            })
+        }
+    }
+}
+
+/// Check the filter for the provided service type for pending work. If a service id is provided
+/// verify it is returned by the filter and only run that handler.
+fn wake_up(
+    filters: &FilterCollection,
+    executor: &JobExecutor,
+    message_sender_factory: &(dyn MessageSenderFactory<Vec<u8>> + 'static),
+    service_type: ServiceType<'static>,
+    service_id: Option<FullyQualifiedServiceId>,
+) {
+    let (filter, handler_factory) = match filters
+        .iter()
+        .find(|(filter, _)| filter.service_types().contains(&service_type))
+    {
+        Some((filter, handler_factory)) => (filter, handler_factory),
+        None => {
+            error!("No filter for serivce type {}", service_type);
+            return;
+        }
+    };
+
+    let service_ids = match filter.filter() {
+        Ok(service_ids) => service_ids,
+        Err(err) => {
+            warn!("Unable to get service IDs from timer filter: {}", err);
+            return;
+        }
+    };
+
+    if let Some(id) = service_id {
+        if service_ids.contains(&id) {
+            let new_handle_factory = handler_factory.clone_box();
+            let msg_sender_factory = message_sender_factory.clone_boxed();
+            executor.execute(move || {
+                let timer_sender = match msg_sender_factory.new_message_sender(&id) {
+                    Ok(timer_sender) => timer_sender,
+                    Err(err) => {
+                        error!("Unable to get message sender: {}", err);
+                        return;
+                    }
+                };
+
+                let mut handler = {
+                    match new_handle_factory.new_handler() {
+                        Ok(handler) => handler,
+                        Err(err) => {
+                            error!(
+                                "Unable to get timer handler for \
+                                    service {}: {}",
+                                id, err
+                            );
+                            return;
+                        }
+                    }
+                };
+
+                if let Err(err) = handler.handle_timer(&*timer_sender, id) {
+                    error!("{}", err);
+                }
+            })
+        } else {
+            warn!(
+                "Received a wake up for service id that doesn't have \
+                pending work: {}",
+                id
+            )
+        }
+    } else {
+        for service_id in service_ids.into_iter() {
+            let new_handle_factory = handler_factory.clone_box();
```

Comment:
Not a blocker but we should choose one of clone_box or clone_boxed and stick with it

Reply:
Story added for the cleanup